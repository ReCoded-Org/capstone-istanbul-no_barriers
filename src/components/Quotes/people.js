import aisha from "./images/Photo-by-Andrea-Piacquadio-from-Pexels.jpg";
import latif from "./images/pexels-justin-shaifer-1222271.jpg";
import şen from "./images/pexels-christina-morillo-1181686.jpg";

const people = [
  {
    photo: aisha,
    quote: "homePage.quotes.quoteAisha",
    job: "homePage.quotes.personAisha",
    id: "homePage.quotes.idAisha",
  },
  {
    photo: latif,
    quote: "homePage.quotes.quoteLatif",
    job: "homePage.quotes.personLatif",
    id: "homePage.quotes.idLatif",
  },
  {
    photo: şen,
    quote: "homePage.quotes.quoteŞen",
    job: "homePage.quotes.personŞen",
    id: "homePage.quotes.idŞen",
  },
];

export default people;
