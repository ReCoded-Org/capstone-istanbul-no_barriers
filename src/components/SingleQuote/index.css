/* quote container with img, paragraph and name */
.quotesQuoteBox {
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
}

.quote {
  font-style: italic;
}

.quotesImg {
  border-radius: 50%;
  height: 4rem;
  width: auto;
  margin: 0.5rem 0;
  display: block;
}
