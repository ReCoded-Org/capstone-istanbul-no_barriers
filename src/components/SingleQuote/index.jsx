import React from "react";
import "./index.css";
import { useTranslation } from "react-i18next";

const SingleQuote = ({ photo, quote, job }) => {
  const { t } = useTranslation();

  return (
    <div className="quotesQuoteBox">
      <q className="quote">{t(quote)}</q>
      <img src={photo} alt="person" className="quotesImg" />
      <p>{t(job)}</p>
    </div>
  );
};

export default SingleQuote;
