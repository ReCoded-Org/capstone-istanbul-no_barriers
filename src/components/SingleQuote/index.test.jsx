import React from "react";
import renderer from "react-test-renderer";
import SingleQuote from "./index";
import people from "../Quotes/people";

it("Single quote renders correctly", () => {
  const tree = renderer
    .create(
      people.map((person) => (
        <SingleQuote
          key={person.id}
          photo={person.photo}
          quote={person.quote}
          job={person.job}
        />
      ))
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
