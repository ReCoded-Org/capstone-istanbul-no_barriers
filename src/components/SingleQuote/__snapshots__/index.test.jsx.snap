// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Single quote renders correctly 1`] = `
Array [
  <div
    className="quotesQuoteBox"
  >
    <q
      className="quote"
    >
      homePage.quotes.quoteAisha
    </q>
    <img
      alt="person"
      className="quotesImg img-fluid"
      src="Photo-by-Andrea-Piacquadio-from-Pexels.jpg"
    />
    <p>
      homePage.quotes.personAisha
    </p>
  </div>,
  <div
    className="quotesQuoteBox"
  >
    <q
      className="quote"
    >
      homePage.quotes.quoteLatif
    </q>
    <img
      alt="person"
      className="quotesImg img-fluid"
      src="pexels-justin-shaifer-1222271.jpg"
    />
    <p>
      homePage.quotes.personLatif
    </p>
  </div>,
  <div
    className="quotesQuoteBox"
  >
    <q
      className="quote"
    >
      homePage.quotes.quoteŞen
    </q>
    <img
      alt="person"
      className="quotesImg img-fluid"
      src="pexels-christina-morillo-1181686.jpg"
    />
    <p>
      homePage.quotes.personŞen
    </p>
  </div>,
]
`;
